.github/workflows/build.yaml

```diff
@@ -1,26 +1,366 @@
+      if: github.event_name == 'push' && github.ref_type == 'branch' && github.ref_name == 'master'
+
+      permissions:
+        pages: write
+        id-token: write
+
+      environment:
+        name: github-pages
+        url: ${{ steps.deployment.outputs.page_url }}
+
+      runs-on: ubuntu-latest
+      needs: docs
+      name: Deploy Docs to GH Pages
+      steps:
+        - name: Deploy
+          id: deployment
+          uses: actions/deploy-pages@v1
+
+    release:
+      # Only run the 'release' job if this workflow was triggered by the creation of a tag
+      if: github.event_name == 'push' && github.ref_type == 'tag'
+
+      runs-on: ubuntu-latest
+      needs: [tarball, docker_builds]
+      name: Create Release
+      steps:
+        - name: Download Tarball
+          uses: actions/download-artifact@v3
+          with:
+            name: tarball
+            path: tarball
+
+        - name: Download VSCode
+          uses: actions/download-artifact@v3
+          with:
+            name: release_vscode
+            path: vscode
+
+        # We don't actually release/upload wake_static. Wake is unsupported on old versions
+        # of alpine, and releasing a "wake_static" artifact is very misleading and prone to
+        # user error and confusion
+        - name: Download Alpine
+          uses: actions/download-artifact@v3
+          with:
+            name: release_alpine
+            path: alpine
+
+        - name: Download Debian Bullseye
+          uses: actions/download-artifact@v3
+          with:
+            name: release_debian_bullseye
+            path: debian_bullseye
+
+        - name: Download CentOS 7.6
+          uses: actions/download-artifact@v3
+          with:
+            name: release_centos_7_6
+            path: centos_7_6
+
+        - name: Download Rocky 8
+          uses: actions/download-artifact@v3
+          with:
+            name: release_rocky_8
+            path: rocky_8
+
+        - name: Download Ubuntu 18.04
+          uses: actions/download-artifact@v3
+          with:
+            name: release_ubuntu_18_04
+            path: ubuntu_18_04
+
+        - name: Download Ubuntu 22.04
+          uses: actions/download-artifact@v3
+          with:
+            name: release_ubuntu_22_04
+            path: ubuntu_22_04
+
+        - name: Rename artifacts
+          run: |
+            TAG="${{ github.ref_name }}"
+            VERSION="${TAG:1}"
+            cp centos_7_6/wake-*-1.x86_64.rpm centos-7-6-wake-${VERSION}-1.x86_64.rpm
+            cp rocky_8/wake-*-1.x86_64.rpm rocky_8-wake-${VERSION}-1.x86_64.rpm
+            cp debian_bullseye/wake_*-1_amd64.deb debian-bullseye-wake_${VERSION}-1_amd64.deb
+            cp ubuntu_18_04/wake_*-1_amd64.deb ubuntu-18-04-wake_${VERSION}-1_amd64.deb
+            cp ubuntu_22_04/wake_*-1_amd64.deb ubuntu-22-04-wake_${VERSION}-1_amd64.deb
+
+        - name: Create Release
+          uses: marvinpinto/action-automatic-releases@latest
```

Comment:
MIT license and yeah this is someone else's action. I reviewed the action code and it looks fine and seems to have a lot of users but I think we should consider keeping our own fork of this

Reply:
Ashley and I spoke offline about this. I'm still in favor of forking but if enough others or if Matthew feels that a commit hash is an acceptable means of pinning the action code then I'm down. If we're not down to fork or if we can't reach consensus on this soon enough we'll just remove this code and fall back to manually uploading like we were before. This would be a nice plus though.

Reply:
Further research reveals that git is using SHA-1DC which is somehow hardened against the SHA1 attacks but is otherwise compatible and at present has no publicly known attacks against it. So a commit hash is probably secure enough for our purposes.

Reply:
Done! PTAL!

Reply:
nvm, just saw the most recent comment. Thanks!